integration/conftest.py

```diff
@@ -154,6 +158,24 @@ def cleanup():
     return furl
 
 
+def _tahoe_runner_optional_coverage(proto, reactor, request, other_args):
+    """
+    Internal helper. Calls spawnProcess with `-m
+    allmydata.scripts.runner` and `other_args`, optionally inserting a
+    `--coverage` option if the `request` indicates we should.
+    """
+    if request.config.getoption('coverage'):
+        args = [sys.executable, '-m', 'coverage', 'run', '-m', 'allmydata.scripts.runner', '--coverage']
+    else:
+        args = [sys.executable, '-m', 'allmydata.scripts.runner']
+    args += other_args
+    return reactor.spawnProcess(
+        proto,
+        sys.executable,
+        args,
+    )
+
+
```

Comment:
I wonder if `_tahoe_runner_optional_coverage` should be merged with `_run_node` somehow.  Unfortunately they do offer slightly different functionality...  If they're not readily mergeable, maybe they can at least share the new coverage logic that is duplicated between them (and in `_create_node` and `util.cli` and maybe other places)?

I also wonder if `_tahoe_runner_optional_coverage` would be better off just accepting a `coverage` boolean parameter instead of the whole request - since that one boolean is all it needs and the API surface of the request object is rather large.  But this idea might be in conflict with the idea of merging with `_run_node` since `_run_node` uses more of the API surface of the request...

Reply:
okay, I moved that to util as well, and have `_run_node` use `_tahoe_runner_optional_coverage`